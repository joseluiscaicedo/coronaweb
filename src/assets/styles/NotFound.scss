@import "./Vars.scss";
.notFound {
  display: grid;
  grid-gap: 1rem;
  place-items: center;
  grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));
  padding: 2rem;
}

.imgContainer__img{
  object-fit: contain;
  width: 100%;
}

.notFound__container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  text-align: center;
  margin-top: 2rem;
  color: $third-color;
  height: 50vh;
}

.container__h1 {
  font-size: 5rem;
  color:$danger-color;
}

.container__h2 {
  font-size: 2rem;
  line-height: 2.4rem;
}

.container__anchor {
  text-decoration: none;
  margin-right: 20px;
  color: inherit;
}

.container__anchor:hover {
  color: $danger-color;
  transition: 2s;
}