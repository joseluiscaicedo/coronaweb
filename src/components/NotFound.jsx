import React from 'react';
import '../assets/styles/NotFound.scss';

const backtoHome = 'https://res.cloudinary.com/josecaicedo/image/upload/v1627426970/coronaweb/backToTheHomepage_txqgtu.png';
const delorean = 'https://res.cloudinary.com/josecaicedo/image/upload/v1627426970/coronaweb/Delorean_pfddqk.webp';

const NotFound = () => (
  <body>
    <section className='notFound'>
      <div className='notFound__img'>
        <img src={backtoHome} alt='Back to the Homepage' />
        <img src={delorean} alt='El Delorean, El Doc y Marti McFly' />
      </div>
      <div className='notFound__container'>
        <h1 className='container__h1'>404</h1>
        <h2 className='container__h2'>PAGE NOT FOUND</h2>
        <h3 className='container__h3'>BACK TO HOME?</h3>
        <a href='/' className='container__anchor'>YES</a>
        <a className='container__anchor' href='https://www.youtube.com/watch?v=G3AfIvJBcGo'>NO</a>
      </div>
    </section>
  </body>
);

export default NotFound;
