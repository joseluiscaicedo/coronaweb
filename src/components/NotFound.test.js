import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import  NotFound from './NotFound';


  test('Validate by HeadingElement', () => {
     const component =render( <NotFound /> );
     const HeadingElement=component.getByText('PAGE NOT FOUND')
    expect(HeadingElement).toBeInTheDocument();
  });

  test('Generates Snapshot for NotFound Component', () => {
    const { container } = render(
      <NotFound />
    );
    expect(container.firstChild).toMatchSnapshot();
  });