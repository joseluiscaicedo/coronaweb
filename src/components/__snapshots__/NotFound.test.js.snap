// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates Snapshot for NotFound Component 1`] = `
<section
  class="notFound"
>
  <div
    class="notFound__imgContainer"
  >
    <img
      alt="Back to the Homepage"
      class="imgContainer__img"
      src="https://res.cloudinary.com/josecaicedo/image/upload/v1627426970/coronaweb/backToTheHomepage_txqgtu.png"
    />
    <img
      alt="El Delorean, El Doc y Marti McFly"
      class="imgContainer__img"
      src="https://res.cloudinary.com/josecaicedo/image/upload/v1627426970/coronaweb/Delorean_pfddqk.webp"
    />
  </div>
  <div
    class="notFound__container"
  >
    <h1
      class="container__h1"
    >
      404
    </h1>
    <h2
      class="container__h2"
    >
      PAGE NOT FOUND
    </h2>
    <h3
      class="container__h3"
    >
      BACK TO HOME?
    </h3>
    <a
      aria-describedby="GoToHome"
      class="container__anchor"
      href="/"
    >
      YES
    </a>
    <a
      aria-describedby="GoOut"
      class="container__anchor"
      href="https://www.youtube.com/watch?v=G3AfIvJBcGo"
    >
      NO
    </a>
  </div>
</section>
`;
